src/basicfuns.jl

```diff
@@ -44,6 +44,31 @@ function xlog1py(x::Number, y::Number)
     return iszero(x) && !isnan(y) ? zero(result) : result
 end
 
+"""
+$(SIGNATURES)
+
+Return `x * exp(x)` for `x > -Inf`, handling `x = -Inf` by taking the downward limit.
+
```

Comment:
Instead of using limits, I would suggest just describing the special case explicitly (`exp(-Inf) == 0.0`).

Reply:
Also updated this docstring